examples/lock-app/efr32/src/LockManager.cpp

```diff
@@ -67,17 +80,20 @@ CHIP_ERROR LockManager::Init(chip::app::DataModel::Nullable<chip::app::Clusters:
 bool LockManager::ReadConfigValues()
 {
     size_t outLen;
-    EFR32Config::ReadConfigValueBin(EFR32Config::kConfigKey_LockUser, reinterpret_cast<uint8_t *>(&mLockUser),
-                                    sizeof(EmberAfPluginDoorLockUserInfo), outLen);
+    EFR32Config::ReadConfigValueBin(EFR32Config::kConfigKey_LockUser, reinterpret_cast<uint8_t *>(&mLockUsers),
+                                    sizeof(EmberAfPluginDoorLockUserInfo) * ArraySize(mLockUsers), outLen);
+
     EFR32Config::ReadConfigValueBin(EFR32Config::kConfigKey_Credential, reinterpret_cast<uint8_t *>(&mLockCredentials),
-                                    sizeof(EmberAfPluginDoorLockCredentialInfo), outLen);
+                                    sizeof(EmberAfPluginDoorLockCredentialInfo) * ArraySize(mLockCredentials), outLen);
 
-    EFR32Config::ReadConfigValueStr(EFR32Config::kConfigKey_LockUserName, mUserName, DOOR_LOCK_USER_NAME_BUFFER_SIZE, outLen);
+    EFR32Config::ReadConfigValueBin(EFR32Config::kConfigKey_LockUserName, reinterpret_cast<uint8_t *>(mUserNames),
+                                    sizeof(mUserNames), outLen);
 
-    EFR32Config::ReadConfigValueBin(EFR32Config::kConfigKey_CredentialData, mCredentialData, sizeof(mCredentialData), outLen);
+    EFR32Config::ReadConfigValueBin(EFR32Config::kConfigKey_CredentialData, reinterpret_cast<uint8_t *>(mCredentialData),
+                                    sizeof(mCredentialData), outLen);
 
-    EFR32Config::ReadConfigValueBin(EFR32Config::kConfigKey_UserCredentials, reinterpret_cast<uint8_t *>(mCredentials.Get()),
-                                    sizeof(DlCredential), outLen);
+    EFR32Config::ReadConfigValueBin(EFR32Config::kConfigKey_UserCredentials, reinterpret_cast<uint8_t *>(mCredentials[0].Get()),
+                                    sizeof(DlCredential) * mMaxUsers * mMaxCredentialsPerUser, outLen);
```

Comment:
This makes no sense.  There are mMaxUsers buffers, each of size mMaxCredentialsPerUser, right?  But this is going to read `mMaxUsers * mMaxCredentialsPerUser` credentials into the _first_ buffer, writing outside its bounds, trashing the heap, and not necessarily filling the other buffers at all.  Unless I am missing something?  What am I missing?

Reply:
I guess I didn't test thoroughly enough here. The members in mCredentials(credentialType and credentialIndex) are only accessed by the getUser command. I was testing this by sending a lock/unlock command after reset. 

Since there's no way to pass a memory offset to the ReadConfigValueBin function, I guess each user will need it's own config key.

Reply:
Addressed in this PR - https://github.com/project-chip/connectedhomeip/commit/b7677b1e29d31806d1a9c3c0f51e483f448641e7